eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -95,6 +115,105 @@ function GetPackageLookup($packageList) {
   return $packageLookup
 }
 
+function create-metadata-table($readmePath, $moniker, $msService, $clientTableLink, $mgmtTableLink, $clientExists, $mgmtExists, $serviceName)
+{
+  New-Item -Path $readmePath -Force
+  $lang = $LanguageDisplayName
+  $langTitle = "Azure $serviceName SDK for $lang"
+  $header = GenerateDocsMsMetadata -language $lang -serviceName $serviceName `
+    -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
+    -msService $msService
+  Add-Content -Path $readmePath -Value $header
+
+  # Add tables, seperate client and mgmt.
+  $readmeHeader = "# $langTitle - $moniker"
+  Add-Content -Path $readmePath -Value $readmeHeader
+  if ($clientExists) {
+    $clientTable = "## Client packages - $moniker`r`n"
+    $clientTable += "[!INCLUDE [client-packages]($clientTableLink)]`r`n"
+    Add-Content -Path $readmePath -Value $clientTable
+  }
+  if ($mgmtExists) {
+    $mgmtTable = "## Management packages - $moniker`r`n"
+    $mgmtTable += "[!INCLUDE [mgmt-packages]($mgmtTableLink)]`r`n"
+    Add-Content -Path $readmePath -Value $mgmtTable
+  }
+}
+
+# Update the metadata table on attributes: author, ms.author, ms.service
+function update-metadata-table($readmePath, $serviceName, $msService)
+{
+  $readmeContent = Get-Content -Path $readmePath -Raw
+  $metadataTable = $readmeContent -replace "---((.*\s+)*)---\s+((.*\s+)*)", '$1'
```

Comment:
Why do a replace instead of just matching for this and the rest content. `$readmeContent -match "---(<metadata>(.*\s+)*)---\s+(<content>(.*\s+)*)"` that way it is a little easier to understand what you are trying to do here.

Reply:
Also does this regex correctly span across newlines?

Reply:
Yes. I have checked the status in debugger.